Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -249,11 +250,10 @@ CGImageRef CGImageCreate(size_t width,
                          const float* decode,
                          bool shouldInterpolate,
                          CGColorRenderingIntent intent) {
-    RETURN_NULL_IF(((provider == nullptr) || ![(NSObject*)provider isKindOfClass:[NSData class]]) || (colorSpace == nullptr));
+    RETURN_NULL_IF(provider == nullptr || colorSpace == nullptr);
 
-    NSData* dataProvider = (__bridge NSData*)provider;
-
-    unsigned char* data = (unsigned char*)[dataProvider bytes];
+    woc::unique_cf<CFDataRef> providerData{ CGDataProviderCopyData(provider) };
```

Comment:
for larger files i'm worried that this is going to be allot of extra overhead.
Since we are pretty much duplicating it now.